src/code/z_skin_awb.c

```diff
@@ -1,13 +1,203 @@
 #include "global.h"
+#include "z64skin.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin_awb/func_80138410.s")
+void func_80138410(Skin* skin) {
+    skin->skeletonHeader = NULL;
+    skin->limbCount = 0;
+    skin->vtxTable = NULL;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin_awb/func_80138424.s")
+/**
+ * Initialises the Vtx buffers used for limb at index `limbIndex`
+ */
+void Skin_InitAnimatedLimb(GameState* gameState, Skin* skin, s32 limbIndex) {
+    s32 i;
+    SkinLimb** skeleton = Lib_SegmentedToVirtual(skin->skeletonHeader->segment);
+    SkinAnimatedLimbData* animatedLimbData =
+        Lib_SegmentedToVirtual(((SkinLimb*)Lib_SegmentedToVirtual(skeleton[limbIndex]))->segment);
+    SkinLimbModif* temp_v0 = Lib_SegmentedToVirtual(animatedLimbData->limbModifications);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin_awb/func_8013859C.s")
+    for (i = 0; i < ARRAY_COUNT(skin->vtxTable->buf); i++) {
+        Vtx* vertices = skin->vtxTable[limbIndex].buf[i];
+        SkinLimbModif* modifEntry;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin_awb/func_80138700.s")
+        for (modifEntry = temp_v0; modifEntry < &temp_v0[animatedLimbData->limbModifCount]; modifEntry++) {
+            SkinVertex* skinVertices = Lib_SegmentedToVirtual(modifEntry->skinVertices);
+            SkinVertex* skinVtxEntry;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin_awb/func_801387D4.s")
+            for (skinVtxEntry = skinVertices; skinVtxEntry < &skinVertices[modifEntry->vtxCount];) {
+                Vtx* vtx = &vertices[skinVtxEntry->index];
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin_awb/func_801388E4.s")
+                skinVtxEntry++;
+
+                vtx->n.flag = 0;
+                vtx->n.tc[0] = skinVtxEntry[-1].s;
+                vtx->n.tc[1] = skinVtxEntry[-1].t;
+                vtx->n.a = skinVtxEntry[-1].alpha;
+            }
+        }
+    }
+}
+
+/**
+ * Initializes a skin skeleton to looping animation, dynamically allocating the frame tables,
+ * and dynamically allocating and initializing the Vtx and SkinLimbVtx buffers for its animated limbs
+ */
+void Skin_Init(GameState* gameState, Skin* skin, SkeletonHeader* skeletonHeader, AnimationHeader* animationHeader) {
+    s32 limbCount;
+    s32 i;
+    SkinLimb** skeleton;
+    SkeletonHeader* virtSkelHeader = Lib_SegmentedToVirtual(skeletonHeader);
```

Comment:
```suggestion
    SkeletonHeader* skelHeader = Lib_SegmentedToVirtual(skeletonHeader);
```
Is the `virt` part necessary? `Lib_SegmentedToVirtual` is very common but this is the first time I've seen that prefix used. But I guess it's to differentiate between this temp and the argument `skeletonHeader`?

Reply:
It's pretty unusual to do this to a skeleton at all.

Reply:
yeah, the point is to differentiate it from `skeletonHeader`. I'm not sure if I like `skelHeader` since it is too similar to `skeletonHeader`. Maybe rename the parameter one (`skeletonHeader`) to `skeletonHeaderSeg`?

Reply:
I think the arg name is more deserving of "skeletonHeader" since its what users will be interacting with when using the function. virtSkelHeader seems fine to me honestly even though unusual

Reply:
skelHeader is also fine if its close to the arg name imo, since they are literally the same thing. doesnt make the code any less confusing

Reply:
Having looked at the rest of the files involved here, I think it'd be better to have a consistent convention for all the virtualisation that occurs.